core/src/commonMain/kotlin/com/xebia/functional/xef/llm/models/chat/ChatCompletionRequestWithFunctions.kt

```diff
@@ -0,0 +1,20 @@
+package com.xebia.functional.xef.llm.models.chat
+
+import com.xebia.functional.xef.llm.models.functions.CFunction
+
+data class ChatCompletionRequestWithFunctions(
```

Comment:
In practice for the Open AI use case, ChatCompletion also deals with functions at the expense of many loose parameters that the remote API interprets one way or another based on which ones you set from the request.
I found this more complex to understand than having two features, chat and chat with functions, and that is why it's modeled as two independent objects. For more context:
https://platform.openai.com/docs/api-reference/chat/create#chat/create-functions
https://platform.openai.com/docs/api-reference/chat/create#chat/create-function_call

All those fields and the returns of the messages in the response get more complex to handle in the client when it's all in one object that can do multiple things.

Reply:
@diesalbla going to resolve this conversation. If we want to attempt to change the model is fine, but we can do it in another PR. Feel free to take a shot at it if you think it can be improved.